test/unit/refresh.js

```diff
@@ -427,110 +427,7 @@ describe('swiftserver:refresh', function () {
       assert(error, 'Should throw an error')
-      assert(error.match('does not conform to swagger specification:'), 'does not conform to swagger specification:')
-    })
-
-    after(function () {
-      runContext.cleanTestDirectory()
-    })
-  })
-
-  describe('Generate scaffolded app from a conforming swagger document with invalid path', function () {
-    var swagger = {
-      'swagger': '2.0',
-      'info': {
-        'version': '0.0.0',
-        'title': '<enter your title>'
-      },
-      'basePath': '/basepath',
-      'paths': {
-        '*': {
-          'get': {
-            'description': 'Gets `Person` objects.'
-          }
-        }
-      }
-    }
-
-    var runContext
-    var error
-
-    before(function () {
-        // Mock the options, set up an output folder and run the generator
-      var spec = {
-        appType: 'scaffold',
-        appName: appName,
-        config: {
-          logger: 'helium',
-          port: 4567
-        }
-      }
-      runContext = helpers.run(path.join(__dirname, '../../refresh'))
-        .inTmpDir(function (tmpDir) {
-          spec.fromSwagger = path.join(tmpDir, 'swagger.json')
-          fs.writeFileSync(spec.fromSwagger, JSON.stringify(swagger))
-        })
-        .withOptions({
-          specObj: spec
-        })
-      return runContext.toPromise().catch(function (err) {
-        error = err.message
-      })
-    })
-
-    it('aborts generator with an error when no resources are found', function () {
-      assert(error, 'Should throw an error')
-      assert(error.match('no resources'), 'no resources')
-    })
-
-    after(function () {
-      runContext.cleanTestDirectory()
-    })
-  })
-
-  describe('Generate scaffolded app from swagger where there are paths but no methods', function () {
-    var swagger = {
-      'swagger': '2.0',
-      'info': {
-        'version': '0.0.0',
-        'title': '<enter your title>'
-      },
-      'basePath': '/basepath',
-      'paths': {
-        'cheese': {
-        }
-      }
-    }
-
-    var runContext
-    var error
-
-    before(function () {
-        // Mock the options, set up an output folder and run the generator
-      var spec = {
-        appType: 'scaffold',
-        appName: appName,
-        config: {
-          logger: 'helium',
-          port: 4567
-        }
-      }
-      runContext = helpers.run(path.join(__dirname, '../../refresh'))
-        .inTmpDir(function (tmpDir) {
-          spec.fromSwagger = path.join(tmpDir, 'swagger.json')
-          fs.writeFileSync(spec.fromSwagger, JSON.stringify(swagger))
-        })
-        .withOptions({
-          specObj: spec
-        })
-      return runContext.toPromise().catch(function (err) {
-        error = err.message
-      })
-    })
-
-    it('aborts generator with an error when no resources are found', function () {
-      assert(error, 'Should throw an error')
-      assert(error.match('no resources'), 'no resources')
```

Comment:
The new validation is much more rigorous and will prevent some situation occurring that were possible with the old validation. In those cases, I have removed the test as is would not add any value.
Tests removed:
Check for invalid path: new validator will not allow an invalid path
Check for no resources (no paths): new validator requires that at least 1 path is present.
Check for no methods (verbs): new validator requires that at least 1 valid verb is specified.